### PR TITLE
[r] Support Ingestion of `SeuratCommands`

### LINKED ISSUE #1946: [r] Ingest Seurat's `commands` slot
Seurat stores command logs in a named list

- names are in the form of `command_name.assay_name`
- e.g. `NormalizeData.RNA`

### PR DESCRIPTION
**Pull request by mojaveazure**
Add support for ingestion of `SeuratCommand` logs. Functionality is provided by new `write_soma.SeuratCommand()` method

Implemented SOMA methods:
- `write_soma.SeuratCommand()`: write a `SeuratCommand` out as a one-dimensional `SOMADataFrame` by encoding the entire `SeuratCommand` as a JSON string

Modified SOMA methods:
- `write_soma.Seurat()`:
  - now creates a collection `uns`
  - if command logs are present, writes them out to `uns/seurat_commands`

New dependencies:
- jsonlite (added as a suggested package)

Notes:
- this PR is a smaller redux of #1944 and #1966 
- this PR does not implement outgestion of `SeuratCommand` logs; outgestion is handled by #2024 

resolves #1946

### CODE REVIEW

File: apis/r/R/write_seurat.R [hunk 1]
```diff
@@ -577,6 +578,130 @@ write_soma.Seurat <- function(
...
+  x,
+  uri = NULL,
+  soma_parent,
+  ...,
+  platform_config = NULL,
+  tiledbsoma_ctx = NULL,
+  relative = TRUE
+) {
+  check_package('SeuratObject', version = .MINIMUM_SEURAT_VERSION())
+  check_package('jsonlite')
```
Comment (eddelbuettel) [inline apis/r/R/write_seurat.R:628]: Please add jsonlite to Suggests: in `DESCRIPTION`

Comment (mojaveazure) [reply apis/r/R/write_seurat.R:628]: 🤦 Can't believe I forgot that, thanks for catching. Added in c5a9bc77658c6a05033bf36b2f5c699f1ebd9604

File: apis/r/R/write_seurat.R [hunk 2]
```diff
@@ -577,6 +578,130 @@ write_soma.Seurat <- function(
...
+          tz = attr(xlist[[i]], 'tzone', exact = TRUE) %||% Sys.timezone()
+        )),
+        .encode_as_char,
+        simplify = FALSE,
+        USE.NAMES = TRUE
+      )
+      xlist[[i]] <- as.character(jsonlite::toJSON(ts, auto_unbox = TRUE))
+    }
+    xlist[[i]] <- .encode_as_char(xlist[[i]])
+  }
```
Comment (aaronwolen) [inline apis/r/R/write_seurat.R:685]: I had a little trouble following this section.  Breaking this down a bit and/or some comments would be really helpful.

**Review by aaronwolen — APPROVED** [review]
One minor comment but otherwise looks good!

File: apis/r/R/write_seurat.R (hunk 2 above)
Comment (mojaveazure) [reply apis/r/R/write_seurat.R]: Comments added in 6b7a8459606e67d58af9bf65c04cadecd48edb88

**Pull request merged**